storage/replicate_queue.go

```diff
@@ -83,12 +90,21 @@ func (rq replicateQueue) shouldQueue(now proto.Timestamp, repl *Replica) (should
 }
 
 func (rq replicateQueue) process(now proto.Timestamp, repl *Replica) error {
-	zone, err := lookupZoneConfig(rq.gossip, repl)
+	// Load the system config.
+	cfg, err := rq.gossip.GetSystemConfig()
 	if err != nil {
 		return err
 	}
 
+	// TODO(marc): shouldn't we be checking whether the range needs to be split?
```

Comment:
I'm pretty sure we should. Was this an oversight?
We check for splits in shouldQueue, but not in process.